### PR TITLE
[patch] add 3 new distance type support for agent-ngt

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <i.can.feel.gravity@gmail.com>

### Description:
add 3 new distance type support
- NormalizedAngle
- NormalizedCosine
- Jaccard

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by rinx** [comment]
https://github.com/vdaas/vald/blob/d262d394f43feaa44818f56275db2a2a477b7e0a/charts/vald/values.yaml#L1250-L1254
please update here.

**Comment by kpango** [comment]
@rinx fixed

**Comment by kpango** [comment]
@rinx  sorry I also updated docs agent-on-docker.md

File: docs/tutorial/agent-on-docker.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ This chapter will use NGT for the core engine of Vald Agent.
       dimension: 784
       # bulk insert chunk size
       bulk_insert_chunk_size: 10
-      # distance_type, which should be "l1", "l2" "angle", "hamming", "cosine", "normalizedangle" or "nomralizedcosine"
+      # distance_type, which should be "l1", "l2" "angle", "hamming", "cosine", "normalizedangle", "nomralizedcosine" or "jaccard"
```
Comment (rinx) [inline docs/tutorial/agent-on-docker.md:85]: thanks, but it's misspelled.
```suggestion
      # distance_type, which should be "l1", "l2" "angle", "hamming", "cosine", "normalizedangle", "normalizedcosine" or "jaccard"
```

File: internal/core/ngt/option.go [hunk 2]
```diff
@@ -94,16 +94,18 @@ func WithDistanceTypeByString(dt string) Option {
...
-	case "hamming":
+	case "hamming", "ham":
 		d = Hamming
 	case "cosine", "cos":
 		d = Cosine
-	case "normalizedangle":
+	case "normalizedangle", "nang", "nangle":
 		d = NormalizedAngle
-	case "normalizedcosine":
+	case "normalizedcosine", "nham", "nhamming":
```
Comment (rinx) [inline internal/core/ngt/option.go:105]: i don't know but "normalized cosine" is same as "normalized hamming"?

Comment (kpango) [reply internal/core/ngt/option.go:105]: sorry it's my mistake, when I open this PR I was drunk :beer:

Comment (kpango) [reply internal/core/ngt/option.go]: @rinx sorry I fixed you pointed out.

**Review by rinx — APPROVED** [review]
seems okay.

**Pull request merged**